api/keto/acl/v1alpha1/write_service.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+package keto.acl.v1alpha1;
+
+import "keto/acl/v1alpha1/acl.proto";
+
+option go_package = "github.com/ory/keto/api/keto/acl/v1alpha1;acl";
+option csharp_namespace = "Ory.Keto.Acl.V1Alpha1";
+option java_multiple_files = true;
+option java_outer_classname = "WatchServiceProto";
+option java_package = "sh.ory.keto.acl.v1alpha1";
+option php_namespace = "Ory\\Keto\\Acl\\V1alpha1";
+
+// The write service to create and delete Access Control Lists.
+service WriteService {
+  // Writes one or more relation tuples in a single transaction.
+  rpc WriteRelationTuples(WriteRelationTuplesRequest) returns (WriteRelationTuplesResponse);
+}
+
+// The request of a WriteService.WriteRelationTuples rpc.
+message WriteRelationTuplesRequest {
+  // The write delta for the relation tuples operated in one single transaction.
+  // Either all actions commit or no change takes effect on error.
+  //
+  // The tuple's etag is used for a consistent read-modify-write request flow.
+  // If the etag is not present, the tuple is not validated against the state
+  // of the serverside tuple and get operated anyway.
+  repeated RelationTupleWriteDelta relation_tuple_deltas = 1;
+}
+
+// Write-delta for a WriteRelationTuplesRequest.
+message RelationTupleWriteDelta {
+  enum Action {
+    // Unspecified.
+    // The `WriteRelationTuples` rpc ignores this
+    // RelationTupleWriteDelta if an action was unspecified.
+    ACTION_UNSPECIFIED = 0;
+
+    // Like INSERT with the exception that if the RelationTuple
+    // already exists performs an UPDATE instead.
+    UPSERT = 1;
+
+    // Insertion of a new RelationTuple.
+    //
+    // The `WriteRelationTuples` rpc errors if the
+    // specified RelationTuple already exists.
+    INSERT = 2;
+
+    // Update of the existing RelationTuple with
+    // the intend to refresh its snapshot token.
+    //
+    // The `WriteRelationTuples` rpc errors if the
+    // specified RelationTuple was not found.
+    UPDATE = 3;
+
+    // Deletion of the RelationTuple.
+    // The `WriteRelationTuples` rpc returns NO error
+    // if the specified RelationTuple was not found.
+    DELETE = 4;
+  }
```

Comment:
This only makes sense if we can identify relation tuples somehow, but I don't think it is really possible. An update essentially means
1. delete the old tuple
2. insert the new one

We can add that as a single operation RPC later on, but for now clients can just do these steps themselves. As the parameters will therefore be different for the action types, we should rather separate this into individual RPCs. What do you think?

Reply:
Yes, e.g. the UPDATE action makes little to no sense with the Tuple model.

I'd like to have a single RPC still for transaction support upfront and that we have less Tuple mutation RPCs as possible.

How about having these actions:
- INSERT - no error if already exists
- DELETE - no error if not found

Later we can add actions like:
- INSERT_NEW - errors tx if already exists
- DELETE_FOUND - errors tx if not found

Reply:
Having a real update that allows deletion and insertion atomically in one call does make sense but is not a hard requirement, so we can add it later. The actions you proposed are perfect and can all be used in one RPC with this action enum.

Reply:
Tracked as #328 
Should we maybe remove the `etag`s and solve everything consistently? Would be a better approach IMO.

Reply:
Yes